mlptensorflow.py

```diff
@@ -0,0 +1,167 @@
+from sklearn.metrics import precision_score
+from sklearn.metrics import recall_score
+from sklearn.metrics import f1_score
+import matplotlib.pyplot as plt
+from sklearn import preprocessing
+from sklearn.model_selection import GridSearchCV
+from sklearn.metrics import classification_report
+from sklearn.decomposition import PCA
+from imblearn.under_sampling import RandomUnderSampler
+from imblearn.over_sampling import RandomOverSampler
+
+data = pd.read_csv('mfcc22_df.csv', index_col=False) 
+labels = data.iloc[:,[-1]]
+labels = labels -1
+
+data = data.drop(labels.columns,axis = 1)
+data.describe()
+
+data = data.to_numpy()
+labels = labels.to_numpy()
+(unique, counts) = np.unique(labels, return_counts=True)
+print(unique, counts) 
+
+indexes = np.where(labels == 1)
+data = np.delete(data, indexes, 0)
+labels = np.delete(labels, indexes, 0)
+labels = np.where(labels<1, labels, labels-1)
+
+(unique, counts) = np.unique(labels, return_counts=True)
+print(unique, counts)
+
+"""##Standarization"""
+
+data +=848.919070 #adding abs min value to whole data just to elimiate negative values
+min_max_scaler = preprocessing.MinMaxScaler()
+data = min_max_scaler.fit_transform(data)
+
+"""##Splitting data
+
+"""
+
+X_train, X_test, y_train, y_test = train_test_split(data, labels, test_size=0.2, random_state=42)
+
+X_train, X_val, y_train, y_val = train_test_split(X_train, y_train, test_size=0.25, random_state=42) 
+
+print("Number of training samples:", X_train.shape[0])
+
+print("Number of testing samples:", X_test.shape[0])
+
+print("Number of val samples:", X_val.shape[0])
+
+print("Number of features:", X_train.shape[1])
+
+"""##Undersampling data or Oversampling - to choose"""
+
+# rus = RandomUnderSampler(random_state=0)
+# rus = RandomOverSampler(random_state=0)
+# X_resampled, y_resampled = rus.fit_resample(X_train, y_train)
+# X_valid_res, y_valid_res = rus.fit_resample(X_valid, y_valid)
+# X_resampled.shape, y_resampled.shape, X_valid_res.shape, y_valid_res.shape
+
+"""## PCA """
+
+pca = PCA()
+X_train_new = pca.fit_transform(data)
+
+explained_variance = pca.explained_variance_ratio_
+
+plt.figure(figsize=(6, 4))
+plt.bar(np.arange(0,len(explained_variance)), explained_variance, alpha=0.5, align='center')
+plt.ylabel('Wariancja')
+plt.xlabel('Główne składowe')
+plt.show()
+
+"""#KERAS
+
+"""
+
+num_classes = 7
+
+keras_model = tf.keras.Sequential([
+                                   
+    tf.keras.Input(shape=(22,)),
+    tf.keras.layers.Dense(128, activation='relu'), 
+    tf.keras.layers.Dense(128, activation='relu'), 
+    tf.keras.layers.Dense(64, activation='relu'),
+    tf.keras.layers.Dense(64, activation='relu'), 
+    tf.keras.layers.Dense(64, activation='relu'), 
+    tf.keras.layers.Dense(num_classes, activation='softmax'),    
+  ])
+
+keras_model.compile(
+    optimizer='adam',
+    loss='sparse_categorical_crossentropy',
+    metrics=['accuracy']
+)
+keras_model.summary()
+
+batch_size = 254
```

Comment:
Dlaczego batch_size jest ustawiony na 254?
Zazwyczaj ustawia się potęgi 2, skąd to jest?